### PR TITLE
[stdlib] Implement pythonic lcm function

### PR DESCRIPTION
**Pull request by bgreni**
Similar changes as #2777 but for the `lcm` function.

Suggested followup https://github.com/modularml/mojo/pull/2777#pullrequestreview-2082274419

### CODE REVIEW

File: stdlib/src/builtin/_math.mojo [hunk 1]
```diff
@@ -267,3 +267,71 @@ fn gcd(*values: Int) -> Int:
...
+    """Computes the least common multiple of two integers.
+
+    Args:
+        m: The first integer.
+        n: The second integer.
+
+    Returns:
+        The least common multiple of the two integers.
+    """
+    return abs((m // gcd(m, n)) * n) if m > n else abs((n // gcd(m, n)) * m)
```
Comment (laszlokindrat) [inline stdlib/src/builtin/_math.mojo:287]: Probably need to check for zero divisor. Please also add a test.
```suggestion
    if d: = gcd(m, n):
        return abs((m // d) * n if m > n else (n // d) * m)
    return 0 
```

**Review by laszlokindrat — APPROVED** [review]
Cool, thank you!

File: stdlib/src/builtin/_math.mojo (hunk 1 above)
Comment (bgreni) [reply stdlib/src/builtin/_math.mojo]: ooooo walrus operator

**Comment by laszlokindrat** [comment]
!sync

**Pull request closed without merging**